src/org/ods/component/ScanWithSnykStage.groovy

```diff
@@ -0,0 +1,88 @@
+package org.ods.component
+
+import org.ods.services.SnykService
+
+class ScanWithSnykStage extends Stage {
+  protected String STAGE_NAME = 'Snyk Security Scan'
+  protected SnykService snyk
+
+  ScanWithSnykStage(def script, IContext context, Map config, SnykService snyk) {
+    super(script, context, config)
+    if (!config.containsKey('failOnVulnerabilities')) {
+      config.failOnVulnerabilities = context.failOnSnykScanVulnerabilities
+    }
+    if (!config.organisation) {
+      config.organisation = context.projectId
+    }
+    if (!config.projectName) {
+      config.projectName = context.componentId
+    }
+    if (!config.buildFile) {
+      config.buildFile = 'build.gradle'
+    }
+    this.snyk = snyk
+  }
+
+  def run() {
+    if (!config.snykAuthenticationCode) {
```

Comment:
This stage should fail without auth code, so the earliest that fails than better.
For this reason, would be better to move this to the constructor and let it fail if no auth code is available?